BuildScript.Nuke/ProjectMetadata.cs

```diff
@@ -1,14 +1,42 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 using Nuke.Common.IO;
 
 #pragma warning disable CS8618
 
-internal class ProjectMetadata
+public class ProjectMetadata
 {
-  public AbsolutePath Path { get; init; }
+  public string Configuration { get; init; }
+  public AbsolutePath ProjectPath { get; init; }
   public string ToolsVersion { get; init; }
   public bool IsMultiTargetFramework { get; init; }
+  public IReadOnlyList<string> TargetFrameworks { get; set; }
   public bool IsSdkProject { get; init; }
-}
+  public IReadOnlyList<string> AssemblyPaths { get; set; }
+
+  public override int GetHashCode () => HashCode.Combine(Configuration, ProjectPath, ToolsVersion, IsMultiTargetFramework, TargetFrameworks,
+      IsSdkProject, AssemblyPaths);
+
+  public override string ToString () =>
+      $"{nameof(Configuration)}: {Configuration}, {nameof(ProjectPath)}: {ProjectPath}, {nameof(ToolsVersion)}: {ToolsVersion}, {nameof(IsMultiTargetFramework)}: {IsMultiTargetFramework}, {nameof(TargetFrameworks)}: {string.Join(",", TargetFrameworks)}, {nameof(IsSdkProject)}: {IsSdkProject}, {nameof(AssemblyPaths)}: {string.Join(",", AssemblyPaths)}";
 
+  public override bool Equals (object? obj)
+  {
+    if (ReferenceEquals(null, obj))
+      return false;
+    if (ReferenceEquals(this, obj))
+      return true;
+    if (obj.GetType() != GetType())
+      return false;
+    return Equals((ProjectMetadata) obj);
+  }
+
+  protected bool Equals (ProjectMetadata other) => Configuration == other.Configuration && ProjectPath.ToString().Equals(other.ProjectPath.ToString())
+                                                                                        && ToolsVersion == other.ToolsVersion
+                                                                                        && IsMultiTargetFramework == other.IsMultiTargetFramework
+                                                                                        && TargetFrameworks.SequenceEqual(other.TargetFrameworks)
```

Comment:
`.SequenceEqual` isn't that great to compare unordered collections, we'll try to find an alternative :)